src/ucp/core/ucp_context.h

```diff
@@ -643,10 +645,27 @@ int ucp_is_scalable_transport(ucp_context_h context, size_t max_num_eps)
     return (max_num_eps >= (size_t)context->config.est_num_eps);
 }
 
+static UCS_F_ALWAYS_INLINE double
+ucp_context_get_est_num_eps(ucp_context_h context, int is_prioritized)
+{
+    return is_prioritized ? context->config.ext.prio_est_num_eps :
```

Comment:
should it be ucs_min(context->config.ext.prio_est_num_eps, context->config.est_num_eps) - in case user passes a smaller value in context->config.est_num_eps ?

Reply:
then it will be set to 1 if user did not specify `ESTIMATED_NUM_EPS` (afaiu, this is the case for many non-mpi users)

Reply:
maybe we can limit the condition to scaled scenarios:
```
    return (context->config.est_num_eps > MIN_SCALED_NUM_EPS) && is_prioritized ? context->config.ext.prio_est_num_eps :
                            context->config.est_num_eps;
```

Reply:
if the user did not specify ESTIMATED_NUM_EPS we can end up estimating the latency for "prioritized" endpoints as higher than "regular" endpoints and IMO it's not what we want

Reply:
how about:
```
static UCS_F_ALWAYS_INLINE double
ucp_context_get_est_num_eps(ucp_context_h context, int is_prioritized)
{
    return (context->config.est_num_eps > context->config.ext.prio_max_num_eps) && is_prioritized ? context->config.ext.prio_max_num_eps :
                            context->config.est_num_eps;
}
```

Reply:
i think it's the same as 
```c
is_prioritized ? ucs_min(context->config.est_num_eps, 
                         context->config.ext.prio_max_num_eps) :
                         context->config.est_num_eps
```